tutorials/Native_distributions_and_local_execution/conveyor.md

```diff
@@ -0,0 +1,187 @@
+    *  [Configuring the Sparkle update engine](https://sparkle-project.org/documentation/customization/).
+* Exploit Windows specific features by:
+    * Injecting markup into the generated `AppxManifest.xml` files that Windows 10/11 use to integrate installed apps with the OS.
+    * Store files in protected directories on Windows that other apps can't access, even if run by the same user.
+    * Request elevation to administrator privileges.
+    * Defining sandbox permissions (defaults to unsandboxed).
+* Exploit Linux specific features by:
+    * Adding services that use systemd to start on boot, run on a schedule, run as separate users, run sandboxed etc. (Tip: you can also package *just* servers this way which can make a nice alternative to Docker).
+    * Controlling your `.desktop` file metadata, your Debian control file metadata, etc.
+    * Adding code to postinst/postrm scripts.
+
+## Tips
+
+1. You can create an unpackaged app for local execution by running `conveyor -Kapp.machines=mac.amd64 make app`, where `mac.amd64` should be replaced with the machine identifier of whatever app you want. You can also use `mac.aarch64` for Apple Silicon, `windows.amd64` and `linux.amd64.glibc`. 
+2. Don't package an über-jar. Some platforms support delta updates and using separate files helps. Also, Windows will avoid downloading files it already has (even on first install) by hard linking to files from other apps and vendors. This is a feature of the [MSIX packaging system](https://conveyor.hydraulic.dev/latest/outputs/#windows).
+3. Use `System.loadLibrary` before trying to unpack native libraries from JARs. Most libraries with native components do this already. Conveyor unpacks native libraries to the app's install directory ahead of time, and discards any that are for the wrong OS/CPU architecture.
+4. The `app.dir` system property is set to the absolute install path, so you can use it to load packaged files you added using the `app.inputs` key.
+5. The `app.version` system property is wired to your `app.version` config key, which is set to the value of your Gradle `version` property.
+6. Make a mistake in your packaging? No big deal - use `app.revision` to make a new version of the app that doesn't affect the version numbers displayed to end users.
+7. Explicit modules (those with a valid `module-info.class` file) are automatically linked into the JVM `modules` file, which is a file format optimized for smaller size and faster startup. They won't be available as JARs on disk as a consequence.
+8. Conveyor will add the Java Accessibility Bridge to your application automatically, to ensure blind users can access the UI.
+9. You can include [a bundled config snippet to improve your JVM configuration in various ways](https://conveyor.hydraulic.dev/latest/stdlib/jvm-clients/).
+10. You can use ProGuard or R8 to obfuscate your app without problems, however Conveyor doesn't automate this. You'll need to integrate it into your build.
```

Comment:
We [recently](https://github.com/JetBrains/compose-jb/pull/2313/files) added this support to our Gradle plugin. We will check if it works with Conveyor, when there will be new compatible versions of Conveyor and/or of Compose 1.2.x.

Also, we will put a link to the ProGuard tutorial, when we write it.